core/src/main/java/com/sequenceiq/cloudbreak/service/filesystem/FileSystemConfigService.java

```diff
@@ -0,0 +1,88 @@
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.model.user.IdentityUser;
+import com.sequenceiq.cloudbreak.common.model.user.IdentityUserRole;
+import com.sequenceiq.cloudbreak.domain.FileSystem;
+import com.sequenceiq.cloudbreak.repository.FileSystemRepository;
+import com.sequenceiq.cloudbreak.service.AuthorizationService;
+
+@Service
+public class FileSystemConfigService {
+
+    private static final String ACCESS_DENIED_MESSAGE = "Access denied!";
+
+    @Inject
+    private FileSystemRepository fileSystemRepository;
+
+    @Inject
+    private AuthorizationService authService;
+
+    public Set<FileSystem> retrievePrivateFileSystems(@Nonnull IdentityUser user) {
+        return fileSystemRepository.findForUser(user.getUserId());
+    }
+
+    public FileSystem create(@Nonnull IdentityUser user, @Nonnull FileSystem fileSystem) {
+        setUserDataRelatedFields(user, fileSystem);
+        authService.hasWritePermission(fileSystem);
+        return fileSystemRepository.save(fileSystem);
+    }
+
+    public FileSystem getPrivateFileSystem(String name, @Nonnull IdentityUser user) {
+        FileSystem fileSystem = fileSystemRepository.findByNameInUser(name, user.getUserId());
+        authService.hasReadPermission(fileSystem);
+        return fileSystem;
+    }
+
+    public FileSystem get(Long id) {
+        FileSystem fileSystem = Optional.ofNullable(fileSystemRepository.findOne(id)).orElseThrow(() -> new AccessDeniedException(ACCESS_DENIED_MESSAGE));
+        authService.hasReadPermission(fileSystem);
+        return fileSystem;
+    }
+
+    public Set<FileSystem> retrieveAccountFileSystems(@Nonnull IdentityUser user) {
+        return user.getRoles().contains(IdentityUserRole.ADMIN)
+                ? fileSystemRepository.findAllByAccount(user.getAccount())
+                : fileSystemRepository.findAllByAccountAndOwner(user.getAccount(), user.getUserId());
+    }
+
+    public void delete(Long id, IdentityUser user) {
+        FileSystem fileSystem = Optional.ofNullable(fileSystemRepository.findOne(id)).orElseThrow(() -> new AccessDeniedException(ACCESS_DENIED_MESSAGE));
+        setUserDataRelatedFields(user, fileSystem);
+        authService.hasWritePermission(fileSystem);
+        fileSystemRepository.delete(id);
+    }
+
+    public void delete(String name, @Nonnull IdentityUser user) {
+        FileSystem fileSystem = Optional.ofNullable(fileSystemRepository.findByNameInAccount(name, user.getAccount(), user.getUserId()))
+                .orElseThrow(() -> new AccessDeniedException(ACCESS_DENIED_MESSAGE));
+        setUserDataRelatedFields(user, fileSystem);
+        authService.hasWritePermission(fileSystem);
+        fileSystemRepository.delete(fileSystem.getId());
+    }
+
+    public FileSystem getPublicFileSystem(String name, @Nonnull IdentityUser user) {
+        Set<FileSystem> foundOnes = fileSystemRepository.findAllPublicByNameAndOwner(name, user.getAccount(), user.getUserId());
```

Comment:
I suggest to use single response instead of `Set` and catch NotUniqueException. Or much better to protect with unique contraint